buildAddOn.py

```diff
@@ -0,0 +1,253 @@
+        print('Only one Archicad version supported with local APIDevKit option')
+        return 1
+
+    # Check platform operating system
+    platformName = None
+    if platform.system () == 'Windows':
+        platformName = 'WIN'
+    elif platform.system () == 'Darwin':
+        platformName = 'MAC'
+
+    # Load config data
+    configFile = open(args.configFile)
+    configData = json.load(configFile)
+    addOnName = configData['addOnName']
+
+    optionalParams = None
+    if 'addOnSpecificCMakeParameters' in configData:
+        optionalParams = configData['addOnSpecificCMakeParameters']
+
+    if args.language:
+        configUpper = [lang.upper() for lang in configData['languages']]
+        languageList = [lang.upper() for lang in args.language]
+        if 'ALL' in languageList:
+            languageList = configUpper
+        else:
+            for lang in languageList:
+                if lang not in configUpper:
+                    print('Language not supported')
+                    return 1
+
+    # Create directory for Build and Package
+    rootFolder = pathlib.Path(__file__).parent.parent.absolute()
+    buildFolder = rootFolder / 'Build'
+    packageRootFolder = buildFolder / 'Package' / addOnName
+    devKitFolderList = {}
+
+    os.chdir (rootFolder)
+
+    if PrepareDirectories(rootFolder, buildFolder, packageRootFolder, devKitFolderList, args, configData, platformName) != 0:
+        return 1
+
+    # At this point, devKitFolderList dictionary has all provided ACVersions as keys
+    # For every ACVersion
+    # If release, build Add-On for all languages with RelWithDebInfo configuration
+    # Else build Add-On with Debug and RelWithDebInfo configurations, without language specified   
+    # In each case, if package creation is enabled, copy the .apx/.bundle files to the Package directory
+    for version in devKitFolderList:
+        devKitFolder = devKitFolderList[version]
+
+        if args.release is True:
+            for languageCode in languageList:
+                if BuildAddOn(rootFolder, buildFolder, devKitFolder, version, configData, platformName, 'RelWithDebInfo', languageCode) != 0:
+                    return 1
+                if args.package:
+                    CopyResultToPackage(packageRootFolder, buildFolder, version, addOnName, platformName, 'RelWithDebInfo', languageCode, True)
+
+        else:
+            if BuildAddOn(rootFolder, buildFolder, devKitFolder, version, configData, platformName, 'Debug') != 0:
+                return 1
+            if BuildAddOn(rootFolder, buildFolder, devKitFolder, version, configData, platformName, 'RelWithDebInfo') != 0:
+                return 1
+            if args.package:
+                CopyResultToPackage(packageRootFolder, buildFolder, version, addOnName, platformName, 'Debug')
+                CopyResultToPackage(packageRootFolder, buildFolder, version, addOnName, platformName, 'RelWithDebInfo')
+
+    # Zip packages
+    if args.package:

```

Comment:
Lets test it with two different types of APIDevKit before considering it done.

Reply:
Tested with 25, 26 and 27